src/Storages/Statistics/StatisticsCountMinSketch.cpp

```diff
@@ -28,36 +28,32 @@ StatisticsCountMinSketch::StatisticsCountMinSketch(const SingleStatisticsDescrip
 
 Float64 StatisticsCountMinSketch::estimateEqual(const Field & val) const
 {
-    if (data_type->isValueRepresentedByNumber())
+    /// Try to convert field to data_type. Converting string to proper data types such as: number, date, datetime, IPv4, Decimal etc.
+    /// Return null if val larger than the range of data_type
+    ///
+    /// For example: if data_type is Int32:
+    ///     1. For 1.0, 1, '1', return Field(1)
+    ///     2. For 1.1, max_value_int64, return null
+    Field val_converted;
+    try
     {
-        /// 'val' maybe number or string, method 'convertFieldToType' will
-        ///     1. convert string to number, date, datetime, IPv4, Decimal etc
-        ///     2. return null if val larger than the range of data_type
-        auto val_converted = convertFieldToType(val, *data_type);
+        val_converted = convertFieldToType(val, *data_type);
         if (val_converted.isNull())
             return 0;
+    }
+    catch (...)
+    {
+        /// If the conversion fails for example, when converting 'not a number' to Int32, return 0
```

Comment:
Can we log something?

Reply:
We definitely should. Swallowing exceptions is too error-prone.

Reply:
I remove the try-catch statment for query will fail in sql validation stage and `convertFieldToType` will throw proper exception.

Reply:
I mean log something in `catch` block so that we can know what is wrong there (actually if you do the convertion here it will print a log for every part, that's why I propose to do the convertion in `Estimator`). We do not have to remove the try-catch.

Reply:
Actually the exception will never be thrown bacause query will fail in sql validation stage which is before estimating statistics.

Reply:
either remove the try ... catch ... or log something in this case. I prefer removing it.